stdlib/pathlib.pyi

```diff
@@ -188,16 +189,16 @@ class Path(PurePath):
     def expanduser(self: Self) -> Self: ...
     def read_bytes(self) -> bytes: ...
     def read_text(self, encoding: str | None = ..., errors: str | None = ...) -> str: ...
-    def samefile(self, other_path: str | bytes | int | Path) -> bool: ...
-    def write_bytes(self, data: bytes) -> int: ...
+    def samefile(self, other_path: StrPath) -> bool: ...
```

Comment:
This just calls `other_path.stat()`, which should also work with Path[bytes]. In case it's not a `Path`, it's passed on to `os.stat()`, which uses `int | StrOrBytesPath`. I therefore think the following would be correct:

```suggestion
    def samefile(self, other_path: StrOrBytesPath | int) -> bool: ...
```

Reply:
`Path[bytes]` isn't a thing — `Path` isn't a generic class. pathlib only works with string paths, unlike `os.path` or functions in `os`

Reply:
I always mix up `Path` and `PathLike`. The annotation should be correct nevertheless.

Reply:
Here' how this method is implemented:

```python
    def samefile(self, other_path):
        st = self.stat()
        try:
            other_st = other_path.stat()
        except AttributeError:
            other_st = self.__class__(other_path).stat()
        return os.path.samestat(st, other_st)
```

Som `other_path` can either be:
- Something with `.stat()` method (assuming `Path`, it can be a new protocol, but I don't think it is worth it right now)
- Something that `self.__class__.__new__` accepts: `StrPath` (== `str | PathLike[str]`)

But, `Path` is a part of `StrPath`, because `Path` is `PathLike[str]`.
That's why `StrPath` seems correct to me. Am I missing something?

Reply:
Ah interesting, I was looking at the 3.9 version of the implementation:

https://github.com/python/cpython/blob/b43496c01a554cf41ae654a0379efae18609ad39/Lib/pathlib.py#L1145-L1154

```python
    def samefile(self, other_path):
        """Return whether other_path is the same or not as this file
        (as returned by os.path.samefile()).
        """
        st = self.stat()
        try:
            other_st = other_path.stat()
        except AttributeError:
            other_st = self._accessor.stat(other_path)
        return os.path.samestat(st, other_st)
```

No need to version guard this. Let's just use the latest version (yours).